starter-code/src/clue.js

```diff
@@ -1,94 +1,122 @@
+    image:        'https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg',
+    occupation:   "Actor"}
+    
+    var mrsPeacock = {
+    first_name:   "Eleanor",
+    last_name:    "Peacock",
+    color:        "blue",
+    description:  "She is from a wealthy family and uses her status and money to earn popularity",
+    age:          36,
+    image:        'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+    occupation:   "Socialité"}
+    
+    var mrMustard = {
+    first_name:   "Jack",
+    last_name:    "Mustard",
+    color:        "yellow",
+    description:  "He is a former football player who tries to get by on his former glory",
+    age:          62,
+    image:        'https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg',
+    occupation:   "Retired Football player"}
+    
+    // Weapons
+    
+    var rope = { name: "rope" , weight: 10}
+    var knife = {name: "knife"  , weight: 8}
+    var candlestick = {name: "candlestick" , weight: 2}
+    var dumbbell ={ name: "dumbbell" ,  weight: 30}
+    var poison= {name: "poison",  weight: 2}
+    var axe = {name:"axe" ,  weight: 15}
+    var bat = {name: "bat" ,weight: 13}
+    var trophy = {name: "trophy" ,  weight: 25}
+    var pistol = {name: "pistol" ,   weight: 20}
+    
+    // Rooms
+    var conservatory = {name: "Conservatory"}
+    var dinningRoom = {name : "Dinning Room"}
+    var kitchen = {name: "Kitchen"}
+    var study= { name: "Study"}
+    var library = {name: "Library"}
+    var billiardRoom = {name: "Billiard Room"}
+    var lounge = {name: "Lounge"}
+    var ballroom = {name: "Ballroom"}
+    var hall = {name: "Hall"}
+    var spa = {name: "Spa"}
+    var livingRoom  = {name: "Living Room"}
+    var observatory = {name: "Observatory"}
+    var theater = {name: "Theater"}
+    var guestHouse ={name: "Guest House"}
+    var patio = {name: "Patio"}
+    
+    
+    // Characters Collection
+    var charactersArray = [];
+    
+    charactersArray.push(mrGreen,drOrchid,profPlum,missScarlet,mrsPeacock,mrMustard)
+    
+    // Rooms' Collection
+    var roomsArray = [];
+    
+    roomsArray.push (rope,knife,candlestick,dumbbell,poison,axe,bat,trophy,pistol)
+    
+    // Weapons Collection
+    var weaponsArray = [];
+    
+    weaponsArray.push (conservatory,dinningRoom,kitchen,study,library,billiardRoom,hall,spa,livingRoom,observatory,theater,guestHouse,patio)
+    
+    function randomSelector(array) {
+        return (array[Math.floor(Math.random() * array.length)]);
+    };
+    randomSelector (charactersArray);
+    
+    function pickMistery () {
+        var randomCharacters = randomSelector (charactersArray);
+        var randomRooms = randomSelector (roomsArray);
+        var randomWeapons = randomSelector (weaponsArray);
+        var misteryArray = [randomCharacters,randomWeapons,randomRooms];
+        return misteryArray;
+    }
+    
+    var misteryEnvelope = pickMistery();
+    
+    function revealMistery() {
```

Comment:
you forgot to pass a parameter to your function- 
```
function revealMistery(myMystery) {
  var result =
    myMystery[0].first_name +
    " " +
    myMystery[0].last_name +
    " killed Mr.Boddy using the " +
    myMystery[1].name +
    " in the " +
    myMystery[2].name +
    "!!!!";
  return result;
}
```
In order to pass the last test, the  items had to appear in a specific order. They wanted the weapon before the room